pallets/investments/src/lib.rs

```diff
@@ -0,0 +1,887 @@
+				}
+				.into_account_truncating();
+				T::Tokens::transfer(
+					info.denomination_currency(),
+					&asset_account,
+					&who,
+					collection.payout_asset_invest,
+					false,
+				)?;
+				T::Tokens::transfer(
+					info.payment_currency(),
+					&asset_account,
+					&who,
+					collection.payout_asset_redeem,
+					false,
+				)?;
+
+				// drain the orders that have been collected
+				orders.retain(|order| !collected.contains(&order.id));
+
+				match collect_type {
+					CollectType::Overflowing => {
+						ActiveOrder::<T>::try_mutate(&asset_id, |total_order| -> DispatchResult {
+							if collection.remaining_asset_invest > T::Amount::zero() {
+								Orders::<T>::try_mutate(
+									&who,
+									&asset_id,
+									|orders| -> DispatchResult {
+										let cur_order_id = OrderId::<T>::get(asset_id.clone());
+										let order = Self::get_order(orders, cur_order_id)?;
+
+										Self::do_update_invest_order(
+											total_order,
+											&who,
+											asset_id.clone(),
+											&info,
+											order,
+											order
+												.invest
+												.checked_add(&collection.remaining_asset_invest)
+												.ok_or(ArithmeticError::Overflow)?,
+										)
+									},
+								)?;
+							};
+
+							if collection.remaining_asset_redeem > T::Amount::zero() {
+								Orders::<T>::try_mutate(
+									&who,
+									&asset_id,
+									|orders| -> DispatchResult {
+										let cur_order_id = OrderId::<T>::get(asset_id.clone());
+										let order = Self::get_order(orders, cur_order_id)?;
+
+										Self::do_update_redeem_order(
+											total_order,
+											&who,
+											asset_id.clone(),
+											&info,
+											order,
+											order
+												.redeem
+												.checked_add(&collection.remaining_asset_redeem)
+												.ok_or(ArithmeticError::Overflow)?,
+										)
+									},
+								)?;
+							};
+
+							Ok(())
+						})?;
+					}
+					CollectType::Closing => {
+						T::Tokens::transfer(
+							info.denomination_currency(),
+							&asset_account,
+							&who,
+							collection.remaining_asset_redeem,
+							false,
+						)?;
+						T::Tokens::transfer(
+							info.payment_currency(),
+							&asset_account,
+							&who,
+							collection.remaining_asset_invest,
+							false,
+						)?;
+					}
+				}
+
+				Ok((collected, collection))
+			},
+		)?;
+
+		Self::deposit_event(Event::OrdersCollected {
+			asset_id,
+			who: who.clone(),
+			processed_orders: collected_ids,
+			collection,
+		});
+
+		// TODO: Actually weight this with collected_ids
+		Ok(().into())
+	}
+
+	pub(crate) fn do_update_invest_order(
+		total_order: &mut TotalOrder<T::Amount>,
+		who: &T::AccountId,
+		asset_id: T::AssetId,
+		info: impl AssetProperties<T::AccountId, Currency = CurrencyOf<T>>,
+		order: &mut OrderOf<T>,
+		amount: T::Amount,
+	) -> DispatchResult {
+		let asset_account = AssetAccount { asset_id }.into_account_truncating();
```

Comment:
Currently, all tokens are stored in the `PoolId` derived account. I think we should keep it this way. The advantage of this is that (once Subscan properly supports token balances, which it should in the short term), you can easily see the entire balance of a pool on one account: the currency it holds, the tranche tokens, and all the collateral NFTs.

Reply:
Makes Sense. I will adapt the asset management trait accordingly.

Reply:
Wait. This is not for the pools to hold. It is basically the "in-between", when the pool has not yet signaled it will want to get investments. I think it makes more sense, to keep the "in-between" in this pallet. 

Wdyt?

Reply:
Nah, you are right. THis should be pools. Although, it will eventually show inconsistent state then about "how much does the pool actually have". As it will contain orders that are not fulfilled, and might never be.

Reply:
What about we transfer the tokens after the pool has signaled how much he wants to fulfill?

Reply:
Isn't that how it currently works? As in, my understanding would be: on epoch execution, the pool pallet mints the executed tranche tokens for investments into the pool account. This is the moment when the pool signals to the investments pallet what the fulfillment was, right?

Reply:
Right. The way it currently works is that the pool account holds tokens for the following three things:

* Tokens transferred _from_ investors as part of an invest or redeem
* Tokens yet to be claimed by investors after a successful investment / redemption
* Tokens that have been repaid by borrowers but are on-hold until an epoch close
* Tokens that are available for borrowers to borrow

The separation between those categories is handled entirely by the internal logic of the pools pallet (and is most of what the logic is about).

We could probably rejigger where some of that is stored so that it's clearer in tools like subscan what the state of the tokens is, but that adds extra transfer logic to lots of places where we currently just update a couple integers.